medmodels/matching/algorithms/classic_distance_models.py

```diff
@@ -1,18 +1,40 @@
 from __future__ import annotations
 
-from typing import Optional
+from typing import TYPE_CHECKING, Literal, Optional
 
-import numpy as np
 import polars as pl
+from sklearn.neighbors import NearestNeighbors
+
+if TYPE_CHECKING:
+    import sys
+
+    if sys.version_info >= (3, 10):
+        from typing import TypeAlias
+    else:
+        from typing_extensions import TypeAlias
 
-from medmodels.matching import metrics
 from medmodels.medrecord.types import MedRecordAttributeInputList
 
+Metric: TypeAlias = Literal[
```

Comment:
Might be a bit confusing to have two different `Metric` TypeAliases here and in `medmodels/matching/metrics.py`

Reply:
Totally right. Removed the other one.